subprojects/packagefiles/utf8proc/meson.build

```diff
@@ -31,3 +34,58 @@ pkg.generate(
 )
 
 install_headers('utf8proc.h')
+
+curl = find_program('curl', required: get_option('tests'))
+
+if get_option('tests').require(curl.found()).allowed()
+
+  _testdefs = {
+    'case'         : '',
+    'custom'       : '',
+    'graphemetest' : 'auxiliary/GraphemeBreakTest.txt',
+    'iterate'      : '',
+    'misc'         : '',
+    'normtest'     : 'NormalizationTest.txt',
+    'printproperty': '',
+    'valid'        : '',
+  }
+
+  if cc.has_header_symbol('wchar.h', 'wcwidth', args: '-D_XOPEN_SOURCE')
+    _testdefs += {'charwidth': ''}
+  endif
+
+  foreach _test, _data : _testdefs
+    if _data == ''
+      _args = []
+      _depends = []
+    else
+      _args = fs.name(_data)
+      _depends = custom_target(
+        _args,
+        build_by_default: false,
+        command: [
+          curl,
+          '--continue-at',
+          '-',
+          '--output', '@OUTPUT@',
+          'https://www.unicode.org/Public/15.1.0/ucd' / _data,
+        ],
+        console: true,
+        output: _args,
+      )
+    endif
+    test(
+      _test,
+      executable(
+        _test,
+        'test/tests.c',
+        'test/@0@.c'.format(_test),
+        dependencies: utf8proc_dep,
+      ),
+      args: _args,
+      depends: _depends,
+      workdir: meson.current_build_dir(),
```

Comment:
is this not default?

Reply:
From my experience, you can't rely on the working directory if not specified: it will vary depending on how the project is compiled (e.g. as subproject or not).